### PR TITLE
Fix: Only URL encode data being passed to URLs (#3652)

### PR DESCRIPTION
**Pull request by ryanmeans**
## Related Issue

Fixes [#3652](https://github.com/external-secrets/external-secrets/issues/3652)

## Checklist

- [x] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [x] All commits are signed with `git commit --signoff`
- [x] My changes have reasonable test coverage
- [x] All tests pass with `make test`
- [x] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

File: pkg/common/webhook/webhook.go [hunk 1]
```diff
@@ -166,7 +180,7 @@ func (w *Webhook) GetWebhookData(ctx context.Context, provider *Spec, ref *esv1b
 		return nil, fmt.Errorf("failed to create request: %w", err)
 	}
 	for hKey, hValueTpl := range provider.Headers {
-		hValue, err := ExecuteTemplateString(hValueTpl, data)
+		hValue, err := ExecuteTemplateString(hValueTpl, rawData)
```
Comment (gusfcarvalho) [inline pkg/common/webhook/webhook.go:183]: Isn't this a breaking change?

File: pkg/provider/webhook/webhook.go [hunk 2]
```diff
@@ -133,7 +133,7 @@ func (w *WebHook) GetSecret(ctx context.Context, ref esv1beta1.ExternalSecretDat
 		return nil, err
 	}
 	// Only parse as json if we have a jsonpath set
-	data, err := w.wh.GetTemplateData(ctx, &ref, provider.Secrets)
+	data, err := w.wh.GetTemplateData(ctx, &ref, provider.Secrets, false)
```
Comment (gusfcarvalho) [inline pkg/provider/webhook/webhook.go:136]: Also, isn't this a breaking change?

File: pkg/common/webhook/webhook.go (hunk 1 above)
Comment (gusfcarvalho) [reply pkg/common/webhook/webhook.go:183]: It is, but I guess what's being sad here is that we don't want to use QueryEscape for Headers...

**Comment by ryanmeans** [comment]
@gusfcarvalho Yeah, I think these are breaking changes in so far as bug fixes are generally breaking changes. 

What do you want to do with this, then? If you look at the test case, for example, currently the body would be resolved as 
```
{"folder": "%2Fmyapp%2Fsecrets"}
```
Since the url encoding happens before it reaches the templating engine, there aren't really any ways to work around this either. The same problem would also exist in header values and the result jsonpath, and I think this change is correct for both of those as well.

**Comment by gusfcarvalho** [comment]
> @gusfcarvalho Yeah, I think these are breaking changes in so far as bug fixes are generally breaking changes.

What do you want to do with this, then? If you look at the test case, for example, currently the body would be resolved as

{"folder": "%2Fmyapp%2Fsecrets"}
Since the url encoding happens before it reaches the templating engine, there aren't really any ways to work around this either. The same problem would also exist in header values and the result jsonpath, and I think this change is correct for both of those as well.

Fair enough :)

**Comment by gusfcarvalho** [comment]
thanks for your contribution @ryanmeans. I'll make sure to release the next version as 0.10.0 and add the appropriate breaking changes

**Review by gusfcarvalho — APPROVED** [review]

**Pull request merged**